bisqapps/shared/domain/src/commonMain/kotlin/network/bisq/mobile/domain/data/persistance/KeyValueStorage.kt

```diff
@@ -0,0 +1,54 @@
+package network.bisq.mobile.domain.data.persistance
+
+import com.russhwolf.settings.Settings
+import com.russhwolf.settings.set
+import network.bisq.mobile.domain.data.model.BaseModel
+
+/**
+ * Multi platform key-value storage ("settings")
+ */
+class KeyValueStorage<T : BaseModel>(
+    private val settings: Settings,
+    private val keyPrefix: String,
+    private val serializer: (t: T) -> String,
+    private val deserializer: (String) -> T
+) : PersistenceSource<T> {
+
+    override suspend fun save(item: T) {
+        settings[keyPrefix + item.id] = serializer(item)
+    }
+
+    override suspend fun saveAll(items: List<T>) {
+        items.forEach { save(it) }
+    }
+
+    override suspend fun get(id: String?): T? {
```

Comment:
Why is id nullable?

Reply:
I was using it for a case where you are saving a single obj of a type but now that I think it doesn't really makes sense and could be error-prone, I'll change it soon.

Reply:
found a better solution, is pushed now